crates/gpui/src/platform/linux/wayland/cursor.rs

```diff
@@ -35,29 +35,40 @@ impl Cursor {
     }
 
     pub fn set_icon(&mut self, wl_pointer: &WlPointer, cursor_icon_name: String) {
+        let mut cursor_icon_name = cursor_icon_name.clone();
         if self.current_icon_name != cursor_icon_name {
             if self.theme.is_ok() {
+                let mut buffer: Option<&CursorImageBuffer>;
+
                 if let Some(cursor) = self.theme.as_mut().unwrap().get_cursor(&cursor_icon_name) {
-                    let buffer: &CursorImageBuffer = &cursor[0];
-                    let (width, height) = buffer.dimensions();
-                    let (hot_x, hot_y) = buffer.hotspot();
+                    buffer = Some(&cursor[0]);
+                } else if let Some(cursor) = self.theme.as_mut().unwrap().get_cursor("default") {
```

Comment:
We should convert these `theme.as_mut().unwrap()` to `if let Ok(theme) = &mut self.theme` on line 40.

Reply:
Ok. Fixed.